CRM/Core/Payment/PayPalImpl.php

```diff
@@ -1062,12 +1062,7 @@ public function invokeAPI($args, $url = NULL) {
     $result = self::deformat($response);
 
     if (curl_errno($ch)) {
-      $e = CRM_Core_Error::singleton();
-      $e->push(curl_errno($ch),
-        0, NULL,
-        curl_error($ch)
-      );
-      return $e;
+      throw new PaymentProcessorException(curl_errno($ch));
```

Comment:
Yeah I guess it could be 

```
throw new PaymentProcessorException(curl_error($ch), curl_errno($ch));
```

Reply:
updated per ^^

Reply:
OK - the errno is now duplicated because I have a feeling just putting it as error_code it might get swallowed & more info rather than less is probably helpful if there is a curl error